v2/metrics/datadog_test.go

```diff
@@ -0,0 +1,187 @@
+package metrics_test
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/grafeas/voucher/v2/metrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	mockAPIKey         = "api-key"
+	mockAppKey         = "app-key"
+	mockClock          = 1096329600
+	testSubmitInterval = 100 * time.Millisecond
+)
+
+func TestDatadogClient_Counter(t *testing.T) {
+	var p datadog.MetricsPayload
+	metrics, _ := newMockedDatadogClient(t, &p)
+
+	metrics.CheckRunStart("diy")
+	metrics.Close()
+
+	require.Len(t, p.Series, 1)
```

Comment:
related to question below: `newMockedDatadogClient` 

How does `p` get populated?

Reply:
or does this value get updated when you call `Close() -> submit()` and `d.metrics.SubmitMetrics` on line 232 update the value of `p`?

Reply:
> does this value get updated when you call `Close() -> submit()` and `d.metrics.SubmitMetrics` on line 232

Yep!  With the extra step of `d.metrics.SubmitMetrics()` sending `POST` to the "[datadog API server](https://github.com/Shopify/voucher/blob/c15d4394954e32c60fbef5efe3de03a5b37fbc31/v2/metrics/datadog.go#L56-L61)", which our test [set to be the "mockDatadog" server](https://github.com/Shopify/voucher/blob/c15d4394954e32c60fbef5efe3de03a5b37fbc31/v2/metrics/datadog_test.go#L181-L183).

`p` gets updated because we passed it to that mock HTTP server.